extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisor.java

```diff
@@ -234,26 +232,24 @@ protected int getTaskGroupIdForPartition(String partitionId)
       partitionIds.add(partitionId);
     }
 
-    Integer intForModulo = extractShardNumberFromShardId(partitionId);
-    if (intForModulo == null) {
-      intForModulo = getHashIntFromShardId(partitionId);
+    BigInteger numberFromShardId = extractShardNumberFromShardId(partitionId);
```

Comment:
Would be nice to include a comment here about why we bother with the BigInteger stuff. It looks strange at first glance, and a reader might wonder if it's required for some reason (it's not, from what you said in your other comment).

Reply:
Added a comment here about why it prefers to use the shard num and why BigInteger is used